.gitignore

```diff
@@ -14,4 +14,6 @@ hs_err_pid*
 .idea/
 *.iml
 */target/**
-*/dependency-reduced-pom.xml
\ No newline at end of file
+*/dependency-reduced-pom.xml
+*/**/*.min.js
+*/**/node_modules
```

Comment:
could just use 
`node_modules`, `target`, `dependency-reduced-pom.xml`.
you want to have a build dir, which you'd want to add.